pkg/envoy/xds/server.go

```diff
@@ -383,19 +386,20 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *logrus.Ent
 
 			responseLog.Debugf("sending xDS response with %d resources", len(resp.Resources))
 
+			versionStr := strconv.FormatUint(resp.Version, 10)
 			out := &envoy_api_v2.DiscoveryResponse{
-				VersionInfo: strconv.FormatUint(resp.Version, 10),
+				VersionInfo: versionStr,
 				Resources:   resources,
 				Canary:      resp.Canary,
 				TypeUrl:     state.typeURL,
-				Nonce:       nonce,
+				Nonce:       versionStr,
```

Comment:
The changes re: the nonce are subtle but important. Please add a comment about it in this commit's message.
Also, in the same commit's message, remove this paragraph, which is not true after your more recent changes:
```
We used to consider a redirect as "realized" after we had asked it to
be created. This commit makes a redirect to be considered "realized"
only after receiving a positive acknowledgement from the proxy.
```

Reply:
DOne.